### ISSUE TITLE
Mac os support

### ISSUE THREAD

**Issue by tothbar** [tixl3d/tixl#32]
the project looks awesome, im just curious if a mac os port comming in the future?

**Comment by Sphynxcolt** [comment]
Of course he plans mac support, but right now, the focus is on windows. No one can tell when.

**Comment by iami0** [comment]
is waiting for Mac version also .
plzzzzz'.

**Comment by pixtur** [comment]
Yes. A MacOS version would be incredibly awesome. This is a long term goal for the project.
I wrote some thoughts on that topic [in the FAQ](https://github.com/still-scene/t3/wiki/faq#so-where-are-the-maclinux-versions).

**Comment by pixtur** [comment]
To be honest, for weg-gl exports cables.gl is probably a better tooll.

**Comment by enzyme69** [comment]
2022 and no Mac M1 version, sadly I actually wanna try ...

**Comment by christo** [comment]
It seems fair to say that there is no timeframe estimate nor commitment to implement cross-platform support ever. 

This is not intended to be a criticism, just an attempt to realistically summarise the way the cost-benefit tradeoff appears now and by extrapolation, how it will continue to appear unless something fundamental changes. Making Tooll better on windows is always going to be an alternative to rewriting core elements for a cross-platform delivery of the same feature set. Doing that work instead is also likely to introduce a permanent additional maintenance and testing overhead. It's not clear what could significantly change the priority of this large undertaking, if anything. 

Put another way, what number of "me too!" comments on this issue would get it on the roadmap for version x.0? Perhaps there's no such number? Would it take a new, committed side team to make it happen? Would it not be easier to make a sister project from scratch that borrows the design paradigm and tries to independently build that alternative framework which would remain at arms-length like Processing and P5.js do?

I offer the above summary to help people who wander in here avoid disappointment and also to avoid that awful train crash of whining comments repeating _still not done!?_ Some people, myself included, may say "never say never" but frankly other people find that overly optimistic and misleading. No dev team likes to feel the negativity that comes from such comments on long-term open feature requests and I wouldn't want the Tooll 3 team to have a polluted experience of the adoration and admiration they clearly deserve.

For the record, Tooll 3 looks fantastic. Well done team. Yes, I wish it would work on Linux and MacOS but my experience tells me not to hold my breath. I'd be delighted to be wrong about this.

Thank you @pixtur for directing those interested in a web-gl  Tooll to consider [cables.gl](https://cables.gl/) instead. That also looks fantastic.

<3

**Comment by pixtur** [comment]
@christo that's a good summary. I am convinced that a cross platform port based on Vulcan is feasible and together with a packaging system would lift Tooll3 onto a new level indeed. It would be definitely break backwards compatibility but I you knew what you're doing, it should be possible to do in a couple of months.

But on the other hand, I KNOW that there is no possible way this could be done with the current team (Which is basically me and a handful of other enthusiasts). My hope is that eventually motivated and experienced Vulcan and .net developers will join the effort to build help with this.

**Comment by reahari** [comment]
@JadeGeek probably not helpful that their tools aren't open source

**Comment by tanis2000** [comment]
> How about game porting toolkit from Apple, it looks like can run DirectX 11 and 12 app very smooth, will you guys check it out? https://developer.apple.com/games/

I will probably check that out once Sonoma gets out of beta. It doesn't really matter whether their tools are open source or not. As an Apple Developer you can access those tools and use them as much as you want. And anyone can become an Apple Developer even without a subscription. It's really not an issue.

**Comment by pixtur** [comment]
That would be sooo awesome. I had a quick look at the game porting toolkit and it actually looks promising.

**Comment by tanis2000** [comment]
I do not have much time at hand but I will probably try that out over the next weekend. I will keep you posted with my findings.

**Comment by tanis2000** [comment]
It looks like 3.7.1 can kind of start but it fails badly when compiling compute shaders when using the game porting toolkit. here is a slice of the log:

```
Loading symbols...
0234:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 0000000001D258A8, 40) stub
026c:fixme:virtual:NtFlushProcessWriteBuffers stub
0234: thread_get_state failed on Apple Silicon - faking zero debug registers
0274: thread_get_state failed on Apple Silicon - faking zero debug registers
0274: thread_get_state failed on Apple Silicon - faking zero debug registers
026c: thread_get_state failed on Apple Silicon - faking zero debug registers
0278: thread_get_state failed on Apple Silicon - faking zero debug registers
0274: thread_get_state failed on Apple Silicon - faking zero debug registers
0274: thread_get_state failed on Apple Silicon - faking zero debug registers
0274: thread_get_state failed on Apple Silicon - faking zero debug registers
Registering loaded symbols...
Applying symbol children...
0234:fixme:ntdll:EtwEventSetInformation (deadbeef, 2, 000000000D674AC8, 46) stub
0274: thread_get_state failed on Apple Silicon - faking zero debug registers
0294: thread_get_state failed on Apple Silicon - faking zero debug registers
0274: thread_get_state failed on Apple Silicon - faking zero debug registers
0294: thread_get_state failed on Apple Silicon - faking zero debug registers
Loading Symbol UIs...
026c: thread_get_state failed on Apple Silicon - faking zero debug registers
026c: thread_get_state failed on Apple Silicon - faking zero debug registers
026c: thread_get_state failed on Apple Silicon - faking zero debug registers
0234: thread_get_state failed on Apple Silicon - faking zero debug registers
Debug: Found no output ui entry for symbol child output 'DataSet' - creating a new one
Warning: Assuming default output position
0274: thread_get_state failed on Apple Silicon - faking zero debug registers
Error: Shader error: Resources\lib\img\internal\render-volume-slice-cs.hlsl:2:1: E5000: syntax error, unexpected NEW_IDENTIFIER
 'render-volume-slice': Resources\lib\img\internal\render-volume-slice-cs.hlsl:2:1: E5000: syntax error, unexpected NEW_IDENTIFIER

Using previous resource state.
Warning: Failed to create compute shader 'render-volume-slice'.
026c: thread_get_state failed on Apple Silicon - faking zero debug registers
Registering Symbol UIs...
Updating UI entries...
Debug: Found no output ui entry for symbol child output 'DataSet' - creating a new one
Warning: Assuming default output position
Creating home...
Error: Shader error: Resources\lib\img\internal\resolve-multisampled-depth-buffer-cs.hlsl:1:1: E5000: syntax error, unexpected KW_TEXTURE2DMS
 'resolve-multisampled-depth-buffer': Resources\lib\img\internal\resolve-multisampled-depth-buffer-cs.hlsl:1:1: E5000: syntax error, unexpected KW_TEXTURE2DMS

Using previous resource state.
Warning: Failed to create compute shader 'resolve-multisampled-depth-buffer'.
0234:err:virtual:virtual_setup_exception stack overflow 2816 bytes in thread 0234 addr 0x6811c52b stack 0x20500 (0x20000-0x21000-0x1a0000)
0288:fixme:thread:NtQueryInformationThread ThreadIsIoPending info class not supported yet
0280:fixme:thread:NtQueryInformationThread ThreadIsIoPending info class not supported yet
0284:fixme:thread:NtQueryInformationThread ThreadIsIoPending info class not supported yet
0294:fixme:thread:NtQueryInformationThread ThreadIsIoPending info class not supported yet
0278:fixme:thread:NtQueryInformationThread ThreadIsIoPending info class not supported yet
0274:fixme:thread:NtQueryInformationThread ThreadIsIoPending info class not supported yet
026c:fixme:thread:NtQueryInformationThread ThreadIsIoPending info class not supported yet
```

Do you happen to know which version of d3dcompiler_xx.dll is required by Tooll3?

**Comment by pixtur** [comment]
Wow. That's already great progress! 
Are you sure, that `0234:err:virtual:virtual_setup_exception stack overflow` is related to the shader pipeline? I'm surprised because the render volume slice shader is an outdated, non-essential experiment. Normally, shaders don't get compiled before usage, which would be after the "Create Home..." line...

Here is what dotpeek says:

(d3dcompiler_47.dll?)

![image](https://github.com/tooll3/t3/assets/1732545/837f59db-b59f-455a-b1a1-b03e5dd7d9bb)

**Comment by tanis2000** [comment]
you are probably right. I believe the shader does not compile and just gets ignored (actually both of them as there are two failing) but then further down the line something else breaks but there is no stack trace to look at. I will try to see if there is any way to get some more debug output

**Comment by zumpchke** [comment]
Can this project state its windows only instead of wasting my time downloading a zip file with .dll and .exe junk.

**Comment by domportera** [comment]
> Can this project state its windows only instead of wasting my time downloading a zip file with .dll and .exe junk.

I was gonna reply with a snarky comment saying "only if you're nice" or "read the install docs - there's no Mac build" but it seems the latter advice would have led you astray by the looks of it - seems like an honest mistake. I'll see if I can do something about that, and if not then @pixtur can

**Comment by pixtur** [comment]
Thanks for looking into that. I can't make up mind of what to be more annoyed of: unfriendly folks on the internet or spam bots.

**Comment by psyfire25** [comment]
I would love to have a go at helping getting this on mac as it looks like a great project. Is anyone activly working on getting it to mac?

**Comment by gfl43r9nd** [comment]
@psyfire25 I don't think so but you can be the one to start ;-)

**Comment by pixtur** [comment]
I think we're (very) slowly making some progress.
The next further building block will be an abstraction layer for Vulcan and then porting the existing DirectX code.

**Comment by mg0x7BE** [comment]
Everyone knows that macOS is a significantly better and nicer system than Windows, so it would be really good if there was a version for macOS since this excellent tool works even on something as hopeless as Windows. We're waiting! :D
